zingolib/src/lightclient/send.rs

```diff
@@ -199,6 +220,61 @@ pub mod send_with_proposal {
             {
                 return Err(CompleteAndBroadcastError::NoSpendCapability);
             }
+            let submission_height = self
+                .get_submission_height()
+                .await
+                .map_err(CompleteAndBroadcastError::SubmissionHeight)?;
+
+            let (sapling_output, sapling_spend): (Vec<u8>, Vec<u8>) = self
+                .read_sapling_params()
+                .map_err(CompleteAndBroadcastError::SaplingParams)?;
+            let sapling_prover = LocalTxProver::from_bytes(&sapling_spend, &sapling_output);
+            let unified_spend_key =
+                UnifiedSpendingKey::try_from(self.wallet.wallet_capability().as_ref())
+                    .map_err(CompleteAndBroadcastError::UnifiedSpendKey)?;
+
+            let mut step_results = Vec::with_capacity(proposal.steps().len());
+            for step in proposal.steps() {
+                let step_result = {
+                    let mut tmamt = self
+                        .wallet
+                        .transaction_context
+                        .transaction_metadata_set
+                        .write()
+                        .await;
+
+                    zcash_client_backend::data_api::wallet::calculate_proposed_transaction(
+                        tmamt.deref_mut(),
+                        &self.wallet.transaction_context.config.chain,
+                        &sapling_prover,
+                        &sapling_prover,
+                        &unified_spend_key,
+                        zcash_client_backend::wallet::OvkPolicy::Sender,
+                        proposal.fee_rule(),
+                        proposal.min_target_height(),
+                        &step_results,
+                        step,
+                    )
+                    .map_err(CompleteAndBroadcastError::Calculation)?
+                };
+
+                step_results.push((step, step_result));
+            }
+            for (_step, step_result) in step_results {
+                self.wallet
+                    .send_to_addresses_inner(
```

Comment:
we need to digest sent transaction BEFORE calculating the second proposal
AND ideally we ougta digest BEFORE broadcasting

Reply:
this is stable or upgrade